### ISSUE TITLE
Error: "Problem while decoding Flate encoded stream: incorrect header check"

### ISSUE THREAD

**Issue by earthlingworks** [gettalong/hexapdf#162]
Hi Thomas! We're seeing a specific error with a PDF: 

`HexaPDF::FilterError (Problem while decoding Flate encoded stream: incorrect header check)`

The script to duplicate:

```
require 'hexapdf'

puts 'File open'
document = HexaPDF::Document.open(ARGV[0])
puts 'File write'
document.write('/tmp/test1', validate: false, optimize: true)

puts 'File 2 open'
document1 = HexaPDF::Document.open('/tmp/test1')
```

I'll send an email with the PDF for you to troubleshoot with. Thanks!

**Comment by gettalong** [comment]
Thanks for the bug report and the file!

I have discovered the problem and along another bug or two and some places where things could be improved. Will fix this and do a release because the original problem is quite severe as it leads to corrupted files.

**Comment by earthlingworks** [comment]
Ok, good to know and thanks!

**Comment by gettalong** [comment]
@earthlingworks I have just released 0.18.0 with the fix!
